usr/src/cmd/iscsid/iscsi-initiator

```diff
@@ -170,6 +171,23 @@ mount_iscsi() {
 }
 
 umount_iscsi () {
+	# In case any non-"zones" zpools reside on iscsi drives, export
+	# all of them now so that the system won't hang later in the shutdown
+	# sequence when those zpools need to be unmounted.
+	zpools=$(zpool list -Ho name)
+	if [ -n $zpools ]; then
+		for pool in $zpools; do
+			if [ "$pool" == "zones" ]; then
```

Comment:
I think this needs to use "svcprop -p config/zpool smartdc/init" (last time something like this came up there was still claims that we supported other system pool names in smartos)